integration_tests/test_store_register.py

```diff
@@ -55,3 +57,15 @@ def test_failed_registration_already_registered(self):
         self.assertThat(str(error.output), Contains(expected))
         self.assertThat(str(error.output), Contains('register-name-dispute'))
 
+    def test_registration_of_reserved_name(self):
+        self.login(expect_success=True)
+        # The snap name is already registered.
+        error = self.assertRaises(
+            subprocess.CalledProcessError,
+            self.register, self.reserved_name)
+        expected = (
+            'The name {0!r} is reserved.\n\n'
+            'If you are the publisher most users expect for {0!r} '
+            'then please claim the name at').format(self.reserved_name)
+        self.assertThat(str(error.output), Contains(expected))
+        self.assertThat(str(error.output), Contains('register-name-dispute'))
```

Comment:
I'm not sure how to make a reserved name in staging.
We can get @nataliabidart to do reserve for us this name, or we can use one that's currently reserved.